File: 왕실의 나이트/sol.c
#include <stdio.h>

int dx[8] = {-2, -2, -1, -1, 1, 1, 2, 2};
int dy[8] = {-1, 1, -2, 2, -2, 2, -1, 1};

int main() {
    int x = getchar() - 'a' + 1;
    int y = getchar() - '0';

    int ans = 0;
    for (int d = 0; d < 8; ++d) {
        if (x + dx[d] >= 1 && x + dx[d] <= 8 && 
            y + dy[d] >= 1 && y + dy[d] <= 8) {
            ans += 1;
        }
    }

    printf("%d\n", ans);

    return 0;
}

File: 왕실의 나이트/sol.py
dx = [-2, -2, -1, -1, 1, 1, 2, 2]
dy = [-1, 1, -2, 2, -2, 2, -1, 1]

x, y = input()
x = int(ord(x)) - int(ord('a')) + 1
y = int(y)

ans = 0
for d in range(8):
    if x + dx[d] >= 1 and x + dx[d] <= 8 and y + dy[d] >= 1 and y + dy[d] <= 8:
        ans += 1

print(ans)